src/main/java/net/library/model/response/UserResponse.java

```diff
@@ -0,0 +1,37 @@
+package net.library.model.response;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import net.library.model.entity.User;
+
+import java.util.List;
+
+public class UserResponse {
+
+    @JsonProperty("x-Correlation-id")
+    private String xCorrelationId;
+
+    @JsonProperty("users")
+    private List<User> users;
```

Comment:
you shouldn't expose the inner structure and return the whole object. Use Dto and return only special fields. these can be all fields or truncated ones